pandas/core/dtypes/common.py

```diff
@@ -1618,6 +1625,13 @@ def is_bool_dtype(arr_or_dtype):
         # this isn't even a dtype
         return False
 
+    if isinstance(arr_or_dtype, (ABCCategorical, ABCCategoricalIndex)):
+        arr_or_dtype = arr_or_dtype.dtype
+
+    if isinstance(arr_or_dtype, CategoricalDtype):
+        arr_or_dtype = arr_or_dtype.categories
+        # now we use the special definition for Index
+
     if isinstance(arr_or_dtype, ABCIndexClass):
```

Comment:
could be elif here

Reply:
Has to be an `if` so that if someone passes a `Categorical` we go `Categorical` -> `Categorical.categories (index)` to this block. We want to go down here since Index has special rules.